pkg/host-disk/host-disk.go

```diff
@@ -88,18 +88,26 @@ func dirBytesAvailable(path string) (uint64, error) {
 	return stat.Bavail * uint64(stat.Bsize), nil
 }
 
-func createSparseRaw(fullPath string, size int64) error {
+func createSparseRaw(fullPath string, size int64) (err error) {
```

Comment:
Personally, I don't like the global error return. I think that it adds some "magic" to the code and makes it harder to read.

Reply:
This is not "global error return but rather named returned value which is a common pattern. 
On long functions this indeed might not be recommended but I think that for such short and clear function it's ok.

Reply:
I think the named return values are mainly used for cases that the defer needs to set some value in them. I think the documentation also describes this pattern.
I saw another usage when initialization of the value is needed.. it may save a few lines of code.

But yes, it really confuses me when reading such code. I guess there is no choice but to get use to it when special cases like these are used.

Reply:
After a lot of investigations I found out that the **only way** to allow defer to change the returned value is to use named returned variables.  Even passing a pointer to the error will not help, I actually looked into the assembler to verify why this is the case . 
I will update the description of the issue so it is clear to  reviewers we must  use named retuned variables we want to check close(). So we can decide if it worth it...